File: rb/features.rb
# ExtinctAnimals SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ExtinctAnimalsFeatures
  def self.make_feature(name)
    case name
    when "base"
      ExtinctAnimalsBaseFeature.new
    when "test"
      ExtinctAnimalsTestFeature.new
    else
      ExtinctAnimalsBaseFeature.new
    end
  end
end

File: py/features.py
# ExtinctAnimals SDK feature factory

from feature.base_feature import ExtinctAnimalsBaseFeature
from feature.test_feature import ExtinctAnimalsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ExtinctAnimalsBaseFeature(),
        "test": lambda: ExtinctAnimalsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
